packages/common/gotrue.ts

```diff
@@ -33,10 +33,29 @@ if (debug) {
   }
 }
 
+let previousOperation: Promise<any> = Promise.resolve()
+
+export async function lockTab<R>(
+  name: string,
+  acquireTimeout: number,
+  fn: () => Promise<R>
+): Promise<R> {
+  try {
+    await previousOperation
+  } catch (e: any) {
```

Comment:
Consider adding error logging in the catch block of the lockTab function. This can help in debugging if any unexpected error occurs. [important]